### DISCUSSION TITLE
Change keywords

### DISCUSSION THREAD

**Discussion by watersprayh** [darmiel/twitch-poll-overlay#23] [Q&A]
Hi,

I don't understand if there is a way to change the keywords

If I want to change them, how should I do it?

Thanks

**Comment by darmiel** [comment]
Hi @watersprayh,

you can change the keywords using the query parameter `reactionJson`.
You can customize the predefined keywords:
```json
[
  {
    "sorting": 0,
    "display": "+1",
    "keywords": ["+1", "1", "yes", "ye", "ja", "sure", "yay"]
  },
  {
    "sorting": 2,
    "display": "~",
    "keywords": ["2"]
  },
  {
    "sorting": 3,
    "display": "-1",
    "keywords": ["-1", "0", "no", "nein", "nay"]
  }
]
```
and then pass them to `reactionsJson`, like this:

https://poll-overlay.d2a.io/?channel=d2aniel&type=bar&timeout=10&requiredPings=3&reactionsJson=%5B%7B%22sorting%22%3A0%2C%22display%22%3A%22%2B1%22%2C%22keywords%22%3A%5B%22%2B1%22%2C%221%22%2C%22yes%22%2C%22ye%22%2C%22ja%22%2C%22sure%22%2C%22yay%22%5D%7D%2C%7B%22sorting%22%3A2%2C%22display%22%3A%22~%22%2C%22keywords%22%3A%5B%222%22%5D%7D%2C%7B%22sorting%22%3A3%2C%22display%22%3A%22-1%22%2C%22keywords%22%3A%5B%22-1%22%2C%220%22%2C%22no%22%2C%22nein%22%2C%22nay%22%5D%7D%5D

You may want to use a [JSON Minifier](https://jsonformatter.org/json-minify) and [URL Encoder](https://www.urlencoder.org/).

**Reply by watersprayh** [reply]
Thank you very much darmiel!

Also, is there any way to show the keyword name within the chart when it is displayed?

**Reply by darmiel** [reply]
Unfortunately, there is currently no way to display a reaction within the chart. 
If the feature is wanted, I can dig into the code again, it's been a few years now 😅.

**Reply by watersprayh** [reply]
If you have the time and inclination it would be very useful for me!
If there would be a way to add a custom label next to the percentage, or instead of it, that would be perfect.

In any case thank you anyway :)

**Reply by darmiel** [reply]
I'm currently busy, I might take a look at it in the next few weeks :)

**Reply by watersprayh** [reply]
Hi @darmiel, any news?
